### PR TITLE
Remove references to autoconf and ./configure

### PR DESCRIPTION
**Pull request by drwells**
I made some other related changes, including:

1. I removed the perl script `lapack_templates.pl` since, to the best of my knowledge, it is not used in the build process anymore.
2. I updated documentation regarding UMFPACK (since it is now built by default).

One thing I cannot figure out is whether or not `DEAL_II_HAVE_TECPLOT` may ever be true. There is no CMake test for this, even though `deal.II` has code that uses tecplot functions.

### CODE REVIEW

**Comment by tamiko** [comment]
No, <code>DEAL_II_HAVE_TECPLOT</code> guards dead code. I do not remember the discussion any more - had we agreed on removing it?

File: contrib/utilities/lapack_templates.pl [hunk 1]
```diff
@@ -1,153 +0,0 @@
...
-	$templates .= "\n\n#ifdef HAVE_S$capname\_";
-	$templates .= "\ninline $type\n$name ($args)\n{\n  s$name\_ ($args2);\n}\n";
-	$templates .= "#else\ninline $type\n$name ($args0)\n";
-	$templates .= "{\n  Assert (false, LAPACKSupport::ExcMissing(\"s$name\"));\n}\n#endif\n";
-    }
-}
-
-print "\n}\n\nDEAL_II_NAMESPACE_OPEN\n";
-
-print "$templates\n\nDEAL_II_NAMESPACE_CLOSE\n\n#endif\n";
```
Comment (tamiko) [inline contrib/utilities/lapack_templates.pl]: Can you remove <code>lapack_templates_custom_target.cmake</code>, <code>lapack_templates.h.in</code> as well?

**Comment by tamiko** [comment]
Otherwise good to go. Thanks for the work!

We can address tecplot in a separate pull-request.

**Comment by bangerth** [comment]
@drwells : Yes, this looks good to me too. I'm glad I didn't scare you off when we finally met this week :-) Please address @tamiko 's comment and rebase to master so that the patch can be merged.

@tamiko : Good question. I don't recall a discussion whether we want to remove tecplot, but given that it's not configured and nobody has complained for the past 2 years since we switched to cmake, it is apparently not a widely used feature.

**Comment by drwells** [comment]
@bangerth : It was great to finally meet you! This is the patch I mentioned (it grew a little).

I removed the changes related to tecplot (since it is not configured, the comments I updated were still incorrect).

**Comment by bangerth** [comment]
Superseded by #484 which contains the rebased version of this patch.

@drwells : For future cases, it is easier if you put like things into separate commits. For example, one commit could have been the lapack_template things, and the adjustments for the ./configure script comments in a separate commit. This often makes it easier to deal with pull requests, and it would make it easier to bring back the lapack_template script (if we ever decided to do so) because then one would only have to revert a single patch without having to figure out which part of the commit is relevant. In any case, though, many thanks for the work here -- much appreciated!

**Comment by drwells** [comment]
@bangerth That's a good point. I was overzealous with my squashing.

**Pull request closed without merging**